ncdfchecker.py

```diff
@@ -376,6 +445,10 @@ def simple_variable_checks(product, constraints, strict=False, logger=None):
                         help="If an unknown variable is encountered or fill "
                         "values specified but not used then it should be "
                         "considered an error.")
+    parser.add_argument('--initdate', '-i',
```

Comment:
I don't see the point in these in order to enforce monthly midpoint dates. If there is a requirement to enforce validity dates outside of the monthly midpoint then this makes sense, otherwise please remove (see comments elsewhere though about needing a config entry if you wish to preserve these)

Reply:
This might actually already be possible through the range controls on the time coordinate anyway, so should probably feed into that as an override and insert entries solution.